tests/utils/test_archive.py

```diff
@@ -32,19 +32,28 @@ def tearDown(self):
         if not debug and self.tmp_dir is not None:
             shutil.rmtree(self.tmp_dir)
 
-    def _write_file(self, fn, contents=None): # pylint: disable=invalid-name
-        full_name = os.path.join(self.tmp_dir, fn)
+    def _write_file(self, filename, contents=None):
+        full_name = os.path.join(self.tmp_dir, filename)
         fileutil.mkdir(os.path.dirname(full_name))
 
-        with open(full_name, 'w') as fh: # pylint: disable=invalid-name
-            data = contents if contents is not None else fn
-            fh.write(data)
+        with open(full_name, 'w') as file_handler:
+            data = contents if contents is not None else filename
+            file_handler.write(data)
             return full_name
 
     @property
     def history_dir(self):
         return os.path.join(self.tmp_dir, 'history')
 
+    @staticmethod
+    def _parse_archive_name(name):
+        # Name can be a directory or a zip
+        # '0000-00-00T00:00:00.000000_incarnation_0'
+        # '0000-00-00T00:00:00.000000_incarnation_0.zip'
+        timestamp_str, _, incarnation_ext = name.split("_")
```

Comment:
Easier way would be just to use `_incarnation_` to split instead of `_` :)

Reply:
D'oh. Thanks! :smile: